modules/tensor_mechanics/include/kernels/StressDivergenceRZTensors.h

```diff
@@ -0,0 +1,42 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef STRESSDIVERGENCERZTENSORS_H
+#define STRESSDIVERGENCERZTENSORS_H
+
+#include "StressDivergenceTensors.h"
+
+//Forward Declarations
+class StressDivergenceRZTensors;
+
+template<>
+InputParameters validParams<StressDivergenceRZTensors>();
+
+/**
+ * StressDivergenceRZTensors is a modification of StressDivergenceTensors to
+ * accommodate the Axisymmetric material models that use cylindrical coordinates.
+ * This kernel is for symmetrical loading only.  The key modifications are a result
+ * of the circumferential stress' dependence on displacement in the axial direction.
+ * Reference: Cook et.al. Concepts and Applications of Finite Element Analysis,
+ * 4th Ed. 2002. p 510.
+ * Within this kernel, '_disp_x' refers to displacement in the radial direction,
+ * u_r, and '_disp_y' refers to displacement in the axial direction, u_z.
+ * The COORD_TYPE in the Problem block must be set to RZ.
```

Comment:
Would it be appropriate to use an if statement to check the coordinate system in both the material and the kernel or only in the kernel?  Similarly, is there a moose method I can employ to check that the RZ kernel is used with the RZ material?